pages/form/Question2.jsx

```diff
@@ -1,78 +1,126 @@
-import { useState } from "react";
+import { useState, useEffect } from "react";
 import {
-  Box,
-  Grid,
+  Wrap,
+  WrapItem,
+  Button,
   Heading,
   Text,
-  Select,
+  Radio,
+  RadioGroup,
+  Collapse,
 } from "@chakra-ui/react";
 import Layout from "../../components/Layout/Layout";
 import useForm from "../../components/FormProvider";
-import { modesOfTransport } from "../../utils/constants";
+import { daysOfWeek, modesOfTransport } from "../../utils/constants";
 import Q2Progress from "../../public/images/progress-bar/q2-progress-dots.svg";
-import { BackButton, ContinueButton } from "../../components/LinkButton/LinkButton";
+import Q2Cloud from "../../public/images/clouds/cloud-q2.svg";
+import LinkButton, { BackButton, ContinueButton } from "../../components/LinkButton/LinkButton";
+
+const WFH = "work from home";
+const ON_SITE = "on-site";
 
 export default function Question2() {
   const { answers, setAnswers } = useForm();
-  const [selectedMode, setSelectedMode] = useState(answers.mainTransportMode || "");
+  const [daysSelected, setDaysSelected] = useState([]);
+  const [workMode, setWorkMode] = useState(answers.workMode);
+  // const [selectedMode, setSelectedMode] = useState(answers.mainTransportMode || "");
+
+  // working around existing backend options "office", "home" and "didNotWork"
+  useEffect(() => {
+    console.table(answers);
+    if (answers.week) {
+      answers.week.forEach((entry, i) => {
+        if (entry === "office") setDaysSelected(prev => [...prev, daysOfWeek[i]]);
+      });
+    }
+  }, [])
 
-  const saveAnswers = () => setAnswers(prev => ({ ...prev, mainTransportMode: selectedMode }));
+  const saveAnswers = () => {
+    setAnswers(prev => ({...prev, workMode: workMode}));
+    const onSiteDays = (workMode === ON_SITE) ? daysSelected : [];
+    // working around existing backend options "office", "home" and "didNotWork"
+    setAnswers(prev => {
+      let week = daysOfWeek.map(day => {
+        if (onSiteDays.includes(day)) return "office";
+        else return "home";
```

Comment:
I think this `else` statement should not be changing elements in week to `home` for days not in `onSiteDays` array.
We are disregarding elements in `week` array that are marked `didNotWork`...

This needs further thought.